src/contour/Audio.cpp

```diff
@@ -0,0 +1,115 @@
+#include <contour/Audio.h>
+
+#include <crispy/assert.h>
+#include <crispy/logstore.h>
+
+#include <qbuffer.h>
+#include <qthread.h>
+
+using namespace contour;
+
+namespace
+{
+constexpr double SAMPLE_RATE = 44100;
+
+// TODO make this function constexpr when we switch to c++23
+double square_wave(double x) noexcept
+{
+    x = std::fmod(x, 2);
+    return std::isgreater(x, 1) ? -1 : 1;
+}
+
+auto createMusicalNote(double volume, double duration, double frequency) noexcept
+{
+    duration = duration / 32.0;
+    volume /= 7;
+    std::vector<std::int16_t> buffer;
+    buffer.reserve(static_cast<size_t>(std::ceil(duration * SAMPLE_RATE)));
+    for (double i = 0; i < duration * SAMPLE_RATE; ++i)
+    {
+        auto sample = static_cast<int16_t>(0x7fff * volume * square_wave(frequency / SAMPLE_RATE * i * 2));
+        buffer.push_back(sample);
+    }
```

Comment:
```suggestion
        buffer.push_back(static_cast<int16_t>(0x7fff * volume * square_wave(frequency / SAMPLE_RATE * i * 2)););
```

just a suggestion. :)